src/main/perf/SearchPerfTest.java

```diff
@@ -190,9 +195,16 @@ private static void _main(String[] clArgs) throws Exception {
     final String fieldName = args.getString("-field");
     final boolean printHeap = args.getFlag("-printHeap");
     final boolean doPKLookup = args.getFlag("-pk");
+    final boolean doConcurrentSegmentReads = args.getFlag("-concurrentSegReads");
     final int topN = args.getInt("-topN");
     final boolean doStoredLoads = args.getFlag("-loadStoredFields");
 
+    int cores = Runtime.getRuntime().availableProcessors();
+    ExecutorService tempExecutorService = new ThreadPoolExecutor(cores, cores, 0L, TimeUnit.MILLISECONDS,
+            new LinkedBlockingQueue<Runnable>(),
+            new NamedThreadFactory("ConcurrentSegReads"));
```

Comment:
Can we rename all of these to `ConcurrentSearching`?

Reply:
I misinterpreted your comment to mean that only the user facing switch to trigger this behaviour needs to be renamed. I will fix all of these now